src/collision.cpp

```diff
@@ -273,68 +272,69 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 
 	bool any_position_valid = false;
 
-	for(s16 x = min.X; x <= max.X; x++)
-	for(s16 y = min.Y; y <= max.Y; y++)
-	for(s16 z = min.Z; z <= max.Z; z++)
-	{
-		v3s16 p(x,y,z);
+	v3s16 p;
+	for (s16 x = min.X; x <= max.X; x++) {
+		p.X = x;
+		for (s16 y = min.Y; y <= max.Y; y++) {
+			p.Y = y;
+			for (s16 z = min.Z; z <= max.Z; z++) {
+				p.Z = z;
```

Comment:
Minor benefit, it unnecessary increases the amount of indents. How about:
`for (p.Z = min.Z; p.z <= max.Z; p.z++) {` ?